app/views/comments/_comment.html.erb

```diff
@@ -29,6 +29,10 @@
                                   commentable.third_user_id == comment.user_id) &&
                                   commentable.decorate.cached_tag_list_array.include?("ama") %>
             <span class="op-marker">ASK ME ANYTHING</span>
+          <% elsif commentable && (commentable.user_id == comment.user_id ||
```

Comment:
I'll start by saying that I think this is way too much logic for a view and it should probably be moved to one of the models or a helper.

Either way, this `elsif` and the `if` above both share a `if commentable` as a first clause so they should probably be wrapped together in a bigger `if` block that checks that in order to DRY the conditionals. 

The `comment_user.id` check is also shared by both blocks and it can be simplified a bit with an `any?` block:

```
elsif [commentable.user_id, commentable.second_user_id, commentable.third_user_id].any? { |id| id == comment.user_id }
```

Put them together and you should be able to replace line 27 through 36 with just:

```
  <% if commentable && [commentable.user_id, commentable.second_user_id, commentable.third_user_id].any? { |id| id == comment.user_id } %>
    <span class="op-marker">
      <%= commentable.decorate.cached_tag_list_array.include?("ama") ? 'ASK ME ANYTHING' : 'OP' %>
    </span>
  <% end %>
```

Reply:
In progress of learning rails so don't know much of rails caveats. Sure I will incorporate the changes you have suggested.

Reply:
@sadarshannaiynar We all gotta start somewhere :) Once you have the basics down, I'd suggest you two books: Eloquent Ruby and Confident Ruby. They both help you learn how to write cleaner Ruby code.

Reply:
@FanaHOVA Thank you. I have also moved the logic from view to helper. Once I am solid with basics I will buy those books. 😄